### PR TITLE
Edit/destroy registration

### PR DESCRIPTION
**Pull request by dandrust**

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -7,7 +7,11 @@ def index
 
   def show
     @event = Event.find(params[:id])
-    @registration = Registration.new(user: current_user, event: @event)
+    if current_user
+      @registration = Registration.first_or_initialize(user: current_user, event: @event)
+    else
+      @registration = Registration.new(user: current_user, event: @event)
```
Comment (Ross-Hunter) [inline app/controllers/events_controller.rb]: Isn't `current_user` nil? Is that still cool?

File: app/controllers/registrations_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class RegistrationsController < ApplicationController
+  before_action :find_registration, only: [:edit, :update, :destroy]
```
Comment (Ross-Hunter) [inline app/controllers/registrations_controller.rb]: 👍

File: app/models/user.rb [hunk 3]
```diff
@@ -36,4 +36,8 @@ def can_lead_event?(event)
     return false unless is_leader
     return event.technology.nil? || qualified_technologies.exists?(event.technology)
   end
+  
+  def registered?(event)
+    return Registration.where(user: self, event: event).present?
```
Comment (Ross-Hunter) [inline app/models/user.rb]: Don't need return here

File: app/views/events/show.html.erb [hunk 4]
```diff
@@ -9,7 +9,7 @@
           <%= f.hidden_field :event_id %>
           <%= f.hidden_field :user_id %>
           <%= f.hidden_field :leader, value: '0' %>
-          <div><%= f.label :name, "Name: #{current_user.name}" %></div>
+          <div><%= f.label :name, "Name: #{current_user.fname} #{current_user.lname}" %></div>
```
Comment (Ross-Hunter) [inline app/views/events/show.html.erb]: Is there a reason you aren't relying on the method in the model?

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (dandrust) [reply app/controllers/events_controller.rb]: Took `user: current_user` out of else case...that's dumb

File: app/models/user.rb (hunk 3 above)
Comment (dandrust) [reply app/models/user.rb]: removed

**Pull request merged**